From: Sven Eckelmann <sven@narfation.org>
Date: Sat, 28 Mar 2026 18:04:03 +0100
Subject: mt7530: Avoid phy errors caused by IRQ rebalancing

Signed-off-by: Sven Eckelmann <sven@narfation.org>

diff --git a/drivers/net/dsa/mt7530.c b/drivers/net/dsa/mt7530.c
--- a/drivers/net/dsa/mt7530.c
+++ b/drivers/net/dsa/mt7530.c
@@ -2227,7 +2227,7 @@ mt7530_setup_irq(struct mt7530_priv *priv)
 		mt7530_set(priv, MT7530_TOP_SIG_CTRL, TOP_SIG_CTRL_NORMAL);
 
 	ret = request_threaded_irq(priv->irq, NULL, mt7530_irq_thread_fn,
-				   IRQF_ONESHOT, KBUILD_MODNAME, priv);
+				   IRQF_ONESHOT | IRQF_NOBALANCING, KBUILD_MODNAME, priv);
 	if (ret) {
 		irq_domain_remove(priv->irq_domain);
 		dev_err(dev, "failed to request IRQ: %d\n", ret);
